app/src/androidTest/java/com/github/se/travelpouch/ui/home/TravelListScreenTest.kt

```diff
@@ -291,5 +298,107 @@ class TravelListScreenTest {
+    dragDistance = 15
+    composeTestRule.onNodeWithTag("mapLatch").performTouchInput {
+      down(center) // Start the drag at the center of the mapLatch
+      moveBy(Offset(0.dp.toPx(), -(dragDistance + correctionFactor).dp.toPx()))
+      up() // End the drag
+    }
+    composeTestRule.onNodeWithTag("mapScreen").assertHeightIsEqualTo(0.dp)
+    composeTestRule.onNodeWithTag("mapScreen").assertIsNotDisplayed()
+    // we will now drag it up by 50 dp, and it shouldn't snap back to 0 dp
+    dragDistance = 50
+    composeTestRule.onNodeWithTag("mapLatch").performTouchInput {
+      down(center) // Start the drag at the center of the mapLatch
+      moveBy(Offset(0.dp.toPx(), +(dragDistance + correctionFactor).dp.toPx()))
+      up() // End the drag
+    }
+    composeTestRule.onNodeWithTag("mapScreen").assertHeightIsEqualTo(dragDistance.dp)
+    composeTestRule.onNodeWithTag("mapScreen").assertIsDisplayed()
+    // we will now drag it up by 60 dp, and this means we're in the zone where the snap to 0 is
+    // available again
+    dragDistance = 60
+    composeTestRule.onNodeWithTag("mapLatch").performTouchInput {
+      down(center) // Start the drag at the center of the mapLatch
+      moveBy(Offset(0.dp.toPx(), +(dragDistance + correctionFactor).dp.toPx()))
+      up() // End the drag
+    }
+    composeTestRule.onNodeWithTag("mapScreen").assertHeightIsEqualTo(110.dp)
+    composeTestRule.onNodeWithTag("mapScreen").assertIsDisplayed()
+    // now it we drag it down again by 20dp, we should hit the threshold and it should close
+    dragDistance = 20
+    composeTestRule.onNodeWithTag("mapLatch").performTouchInput {
+      down(center) // Start the drag at the center of the mapLatch
+      moveBy(Offset(0.dp.toPx(), -(dragDistance + correctionFactor).dp.toPx()))
+      up() // End the drag
+    }
+    composeTestRule.onNodeWithTag("mapScreen").assertHeightIsEqualTo(0.dp)
+    composeTestRule.onNodeWithTag("mapScreen").assertIsNotDisplayed()
+    // now we drag it up and it shouldn't change anything
+    dragDistance = 20
+    composeTestRule.onNodeWithTag("mapLatch").performTouchInput {
+      down(center) // Start the drag at the center of the mapLatch
+      moveBy(Offset(0.dp.toPx(), -(dragDistance + correctionFactor).dp.toPx()))
+      up() // End the drag
+    }
+    composeTestRule.onNodeWithTag("mapScreen").assertHeightIsEqualTo(0.dp)
+    composeTestRule.onNodeWithTag("mapScreen").assertIsNotDisplayed()
+    // and now we don't have any "drag" debt so it we drag it back by 60 dp, it should open
+    dragDistance = 60
+    composeTestRule.onNodeWithTag("mapLatch").performTouchInput {
+      down(center) // Start the drag at the center of the mapLatch
+      moveBy(Offset(0.dp.toPx(), +(dragDistance + correctionFactor).dp.toPx()))
+      up() // End the drag
+    }
+    composeTestRule.onNodeWithTag("mapScreen").assertHeightIsEqualTo(60.dp)
+    composeTestRule.onNodeWithTag("mapScreen").assertIsDisplayed()
```

Comment:
I don't understand this part. How the map can be visible if the height is below the threshold which is 100

Reply:
We discussed about it. Follwing your logic, it is a valid piece of testing